spec/queries/overdue_patients_query_spec.rb

```diff
@@ -152,4 +152,147 @@
         group_by: :called_by_user_id)).to eq(expected)
     end
   end
+
+  context "count_contactable_patients_called" do
+    it "should return count of contactable overdue patients called" do
+      facility = create(:facility)
+      _user_1 = create(:user, registration_facility: facility)
+      _user_2 = create(:user, registration_facility: facility)
+
+      patient_1 = create(:patient, assigned_facility: facility, phone_numbers: [])
+      patient_2 = create(:patient, assigned_facility: facility)
+      patient_3 = create(:patient, assigned_facility: facility)
+      patient_4 = create(:patient, assigned_facility: facility, phone_numbers: [])
+
+      create(:call_result, patient: patient_1, device_created_at: this_month + 15.days)
+      create(:call_result, :remove_from_overdue_list, patient: patient_2, device_created_at: this_month + 1.days)
+      create(:call_result, patient: patient_3, device_created_at: this_month + 27.days)
+      create(:call_result, patient: patient_4, device_created_at: this_month - 4.days)
+
+      RefreshReportingViews.refresh_v2
+
+      previous_month = Date.today - 1.month
+      expected = {
+        Period.month("#{Date::MONTHNAMES[previous_month.month]} #{Date.today.year}") => 2
+      }
+
+      expect(described_class.new.count_contactable_patients_called(facility.region, :month)).to eq(expected)
+    end
+
+    it "should return count of contactable overdue patients called grouped by the group_by condition" do
+      facility = create(:facility)
+      user_1 = create(:user, registration_facility: facility)
+      user_2 = create(:user, registration_facility: facility)
+
+      patient_1 = create(:patient, assigned_facility: facility, phone_numbers: [])
+      patient_2 = create(:patient, assigned_facility: facility)
+      patient_3 = create(:patient, assigned_facility: facility)
+      patient_4 = create(:patient, assigned_facility: facility)
+
+      _call_result_1 = create(:call_result, patient: patient_1, user: user_1, device_created_at: this_month + 15.days)
+      _call_result_1 = create(:call_result, patient: patient_2, user: user_1, device_created_at: this_month + 1.days)
+      _call_result_1 = create(:call_result, patient: patient_3, user: user_1, device_created_at: this_month + 27.days)
+      _call_result_1 = create(:call_result, patient: patient_4, user: user_2, device_created_at: this_month + 4.days)
```

Comment:
[non-blocking] All the call result has same name

Reply:
Ah! My bad